memote/support/biomass.py

```diff
@@ -176,14 +180,56 @@ def find_direct_metabolites(model, reaction):
         that are taken up to be consumed by the biomass reaction only.
 
     """
-    transport_reactions = set(helpers.find_transport_reactions(model))
-    exchange_reactions = set(model.exchanges)
-    biomass_reactions = set(helpers.find_biomass_reaction(model))
-
-    combined_set = transport_reactions | exchange_reactions | biomass_reactions
+    biomass_rxns = set(helpers.find_biomass_reaction(model))
+    tra_bou_bio_rxns = helpers.find_tra_bou_bio_reactions(model, biomass_rxns)
     precursors = find_biomass_precursors(model, reaction)
+    tra_bou_bio_mets = [met for met in precursors if
+                        met.reactions.issubset(tra_bou_bio_rxns)]
+    rxns_of_interest = set([rxn for met in tra_bou_bio_mets
+                            for rxn in met.reactions
+                            if rxn not in biomass_rxns])
+
+    solution = model.optimize()
+    if solution.objective_value == 0:
+        raise ValueError("objective value is zero. Model will not grow.")
 
-    return [met for met in precursors if met.reactions.issubset(combined_set)]
+    tra_bou_bio_fluxes = solution.fluxes[get_ids(rxns_of_interest)]
+    flux_sum = pd.DataFrame(index=tra_bou_bio_mets, columns=["sum"], data=0).T
+
+    main_comp = helpers.find_compartment_id_in_model(model, 'c')
+    exc_space = helpers.find_compartment_id_in_model(model, 'e')
+    for met in tra_bou_bio_mets:
+        for rxn in met.reactions:
+            not_biomass_rxn = rxn not in biomass_rxns
+            extracellular = met.compartment == exc_space
+            # if reactant metabolite in "e"
+            if met in rxn.reactants and not_biomass_rxn and extracellular:
+                product_comps = set([p.compartment for p in rxn.products])
+                # e reactant, no product (outward flux)
+                if len(product_comps) == 0:
+                    flux_sum[met] += -float(
+                        tra_bou_bio_fluxes[get_ids([rxn])])
+                # e reactant, c product (inward flux)
+                elif main_comp in product_comps:
+                    flux_sum[met] += float(
+                        tra_bou_bio_fluxes[get_ids([rxn])])
+            # if product metabolite in "e"
+            elif met in rxn.products and not_biomass_rxn and extracellular:
+                reactant_comps = set([p.compartment for p in rxn.reactants])
+                # e product, no reactant (inward flux)
+                if len(reactant_comps) == 0:
+                    flux_sum[met] += float(
+                        tra_bou_bio_fluxes[get_ids([rxn])])
+                # e product, c reactant (outward flux)
+                elif main_comp in reactant_comps:
+                    flux_sum[met] += -float(
+                        tra_bou_bio_fluxes[get_ids([rxn])])
+            # for all non-e metabolites
+            elif not_biomass_rxn and not extracellular:
+                flux_sum[met] += np.abs(
```

Comment:
According to CodeCov: https://codecov.io/gh/opencobra/memote/compare/f7dbca89ddb5406b2ba4566a88e341c16ad5c051...ab15e59a0311be476099ce7181ff35bfc413cd64/diff

The control flow statements from lines 206 to 225 are not covered in any tests (which is also why the build fails)